src/UnitOfWork.php

```diff
@@ -3019,36 +3014,26 @@ public function getScheduledCollectionUpdates(): array
      */
     public function initializeObject(object $obj): void
     {
-        if ($obj instanceof InternalProxy) {
-            $obj->__load();
-
-            return;
-        }
-
         if ($obj instanceof PersistentCollection) {
             $obj->initialize();
 
             return;
         }
 
-        if ($this->em->getConfiguration()->isNativeLazyObjectsEnabled()) {
-            $reflection = $this->em->getClassMetadata($obj::class)->getReflectionClass();
-            $reflection->initializeLazyObject($obj);
-        }
+        $reflection = $this->em->getClassMetadata($obj::class)->getReflectionClass();
+        $reflection->initializeLazyObject($obj);
     }
 
     /**
      * Tests if a value is an uninitialized entity.
-     *
-     * @phpstan-assert-if-true InternalProxy $obj
      */
     public function isUninitializedObject(mixed $obj): bool
```

Comment:
v4 would be a good chance to use an `object` type hint here

Reply:
Apparently it has to do with https://github.com/doctrine/orm/commit/5e76f120007bc7267005d1fdd14d5cc812975f4d (the values inside `invalidAssociationValuesDataProvider` are the reasons why the test suite fails when switching to `object`)